library/src/scripts/flyouts/NewPostMenu.tsx

```diff
@@ -59,47 +64,302 @@ function ActionItem({ item, style }: { item: IAddPost; style: ITransition }) {
+const reducer = (state, action) => {
+    switch (action.type) {
+        case "toggle_open":
+            return { ...state, open: !state.open };
+        case "set_open":
+            return { ...state, open: action.open };
+        case "set_button_focus":
+            return { ...state, buttonFocus: action.focus };
+        case "set_focused_item":
+            return { ...state, focusedItem: action.item };
+        default:
+            throw new Error();
+    }
+};
+
 export default function NewPostMenu(props: { items: IAddPost[] }) {
-    const [open, setOpen] = useState(false);
-    const toggle = () => setOpen(!open);
-    const onClickBackground = () => {
-        if (open) {
-            setOpen(false);
+    const classes = newPostMenuClasses();
+    const vars = newPostMenuVariables();
+
+    let { items } = props;
+
+    const buttonRef = useRef<HTMLButtonElement>(null);
+    const backgroundRef = useRef<HTMLElement>(null);
+    const menuRef = useRef<HTMLUListElement>(null);
+    const itemsRef = useRef<HTMLLIElement>(null);
+
+    const accessMenuRef = useRef<HTMLUListElement>(null);
+
+    const [state, dispatch] = useReducer(reducer, initialState);
+
+    const onClickBackground = (event: React.MouseEvent) => {
+        event.stopPropagation();
+        if (state.open) {
+            dispatch({ type: "set_button_focus", focus: true });
+            dispatch({ type: "set_open", open: false });
+            dispatch({ type: "set_focused_item", item: undefined });
         }
     };
 
-    const classes = newPostMenuClasses();
-    const { items } = props;
+    const onBgKeyDown = (event: React.KeyboardEvent<any>) => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        switch (event.key) {
+            case "Escape":
+                if (state.open) {
+                    dispatch({ type: "set_open", open: false });
+                    dispatch({ type: "set_button_focus", focus: true });
+                    dispatch({ type: "set_focused_item", item: undefined });
+                }
+                break;
+            case "Home":
+                if (state.open) {
+                    dispatch({ type: "set_focused_item", item: 0 });
+                }
+                break;
+            case "End":
+                if (state.open) {
+                    dispatch({ type: "set_focused_item", item: items.length - 1 });
+                }
+                break;
+            case "ArrowUp":
+                if (state.open && typeof state.focusedItem != "undefined") {
+                    dispatch({ type: "set_focused_item", item: (state.focusedItem + 1) % items.length });
+                }
+                break;
+            case "ArrowDown":
+                if (state.open && typeof state.focusedItem != "undefined") {
+                    dispatch({ type: "set_focused_item", item: (state.focusedItem - 1 + items.length) % items.length });
+                }
+                break;
+            case "A":
```

Comment:
I think you should just use the default here and just search for whatever key. Other languages may have accents or other characters that are not in this set.

Reply:
Got it! Will change!